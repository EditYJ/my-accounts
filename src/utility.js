export const LIST_VIEW = "list";
export const CHART_VIEW = "chart";
export const TYPE_INCOME = "income";
export const TYPE_OUTCOME = "outcome";

export const addZero = month => {
  return month < 10 ? "0" + month : month;
};

export const range = (size, startAt = 0) => {
  const arr = [];
  for (let i = 0; i < size; i++) {
    arr[i] = i + startAt;
  }
  return arr;
};

export const viewActive = (id, selectedId) => {
  return id === selectedId ? "dropdown-item active" : "dropdown-item";
};

export const parseToYearAndMounth = str => {
  const date = str ? new Date(str) : new Date();

  return {
    year: date.getFullYear(),
    month: date.getMonth() + 1
  };
};
