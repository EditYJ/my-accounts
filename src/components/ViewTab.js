/* eslint-disable jsx-a11y/anchor-is-valid */
import React from "react";
import Ionicon from "react-ionicons";
import PropTypes from "prop-types";
import { LIST_VIEW, CHART_VIEW } from "../utility";

const generateLinkClass = (current, view) => {
  return current === view ? "nav-link active" : "nav-link";
};

const ViewTab = ({ activTab, onTabChange }) => {
  return (
    <ul className="nav nav-tabs nav-fill my-2">
      <li className="nav-item">
        <a
          className={generateLinkClass(activTab, LIST_VIEW)}
          href="#"
          onClick={event => {
            event.preventDefault();
            onTabChange(LIST_VIEW);
          }}
        >
          <Ionicon
            className="mr-2"
            fontSize="25px"
            color={"#007bff"}
            icon="ios-paper"
          />
          列表模式
        </a>
      </li>
      <li className="nav-item">
        <a
          className={generateLinkClass(activTab, CHART_VIEW)}
          href="#"
          onClick={event => {
            event.preventDefault();
            onTabChange(CHART_VIEW);
          }}
        >
          <Ionicon
            className="mr-2"
            fontSize="25px"
            color={"#007bff"}
            icon="ios-pie"
          />
          图表模式
        </a>
      </li>
    </ul>
  );
};

ViewTab.propTypes = {
  activTab: PropTypes.string.isRequired,
  onTabChange: PropTypes.func.isRequired
};

ViewTab.defaultProps = {
  onTabChange: view => {
    console.log(view);
  }
};

export default ViewTab;
