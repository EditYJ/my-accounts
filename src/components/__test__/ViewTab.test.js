import React from "react";
import { mount } from "enzyme";
import ViewTab from "../ViewTab";
import { LIST_VIEW,CHART_VIEW } from "../../utility";

const props = {
  activTab: CHART_VIEW,
  onTabChange: jest.fn()
}

let wrapper
describe('ViewTab test',()=>{
  beforeEach(() => {
    wrapper = mount(<ViewTab {...props} />);
  });
  
  it("should render the component to match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('test onTabChange',()=>{
    const firstViewTab = wrapper.find(".nav-item").first()
    const lastViewTab = wrapper.find(".nav-item").last()

    firstViewTab.find('a').simulate('click')
    expect(props.onTabChange).toHaveBeenCalledWith(LIST_VIEW)

    lastViewTab.find('a').simulate('click')
    expect(props.onTabChange).toHaveBeenCalledWith(CHART_VIEW)

    expect(lastViewTab.find('a').hasClass('active')).toBeTruthy()
    expect(firstViewTab.find('a').hasClass('active')).toBeFalsy()

  })
})