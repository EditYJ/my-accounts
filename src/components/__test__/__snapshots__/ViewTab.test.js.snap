// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewTab test should render the component to match snapshot 1`] = `
<ViewTab
  activTab="chart"
  onTabChange={[MockFunction]}
>
  <ul
    className="nav nav-tabs nav-fill my-2"
  >
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="#"
        onClick={[Function]}
      >
        <Ionicon
          beat={false}
          className="mr-2"
          color="#007bff"
          fontSize="25px"
          icon="ios-paper"
          rotate={false}
          shake={false}
          style={Object {}}
        >
          <styled.svg
            beat={0}
            className="mr-2"
            fill="#007bff"
            height="25px"
            rotate={0}
            shake={0}
            style={Object {}}
            viewBox="0 0 1024 1024"
            width="25px"
          >
            <svg
              className="mr-2 sc-bdVaJa fUuvxv"
              fill="#007bff"
              height="25px"
              rotate={0}
              style={Object {}}
              viewBox="0 0 1024 1024"
              width="25px"
            >
              <path
                d="M224 96v736h-32v-608h-64v642.2c0 34 27.6 61.8 61.8 61.8h644.6c34 0 61.8-27.6 61.8-61.8v-770.2h-672.2zM320 224h256v32h-256v-32zM320 544h384v32h-384v-32zM800 736h-480v-32h480v32zM800 416h-480v-32h480v32z"
              />
            </svg>
          </styled.svg>
        </Ionicon>
        列表模式
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link active"
        href="#"
        onClick={[Function]}
      >
        <Ionicon
          beat={false}
          className="mr-2"
          color="#007bff"
          fontSize="25px"
          icon="ios-pie"
          rotate={false}
          shake={false}
          style={Object {}}
        >
          <styled.svg
            beat={0}
            className="mr-2"
            fill="#007bff"
            height="25px"
            rotate={0}
            shake={0}
            style={Object {}}
            viewBox="0 0 1024 1024"
            width="25px"
          >
            <svg
              className="mr-2 sc-bdVaJa fUuvxv"
              fill="#007bff"
              height="25px"
              rotate={0}
              style={Object {}}
              viewBox="0 0 1024 1024"
              width="25px"
            >
              <path
                d="M544 128v448l-395.51 98.752c54.816 165.69 211.192 285.248 395.26 285.248 229.75 0 416.25-186.25 416.25-416s-186-416-416-416v0z M512 64h-11c-320 0-437 232.5-437 414.574 0 0 0.5 113.332 37.82 176.582l410.18-104.010v-487.146z"
              />
            </svg>
          </styled.svg>
        </Ionicon>
        图表模式
      </a>
    </li>
  </ul>
</ViewTab>
`;
